src/container/Posts.js

```diff
@@ -4,14 +4,16 @@ import { connect } from 'react-redux';
 import BlogEntries from '../components/BlogEntries';
 import '../css/mainPage.css';
 import { fetchEntries } from '../redux/actions/blogActions';
+import { Redirect } from 'react-router-dom';
 
 class MainPage extends Component {
   componentDidMount() {
     this.props.fetchEntries();
   }
 
   render() {
-    const blogs = this.props.blogs;
+    const { blogs, auth } = this.props;
+    if (!auth.uid) return <Redirect to="/login" />;
```

Comment:
Good try. I have a suggestion.

* create a higher order component which, checks if you are logged in
* if you are, then it renders the component you gave it
* if you are not, it redirects you to login
* put that HOC, with Posts as a parameter, on the route

In this way, if you have multiple protected routes, you don't copy paste this code everywhere.

Reply:
done